File: CIN.js
const UniqueItem = require("./UniqueItem");
const UtilsMethod = require("./UtilsMethod");
const prompt = require("prompt-sync")();

class CIN extends UniqueItem {
  constructor(lastName, firstName, dateOfBirth, placeOfBirth, idNumber, name) {
    super(name);
    this.lastName = lastName;
    this.firstName = firstName;
    this.dateOfBirth = dateOfBirth;
    this.placeOfBirth = placeOfBirth;
    this.idNumber = idNumber;
  }

  static createFromScanner() {
    console.log("Entrez le prénom : ");
    const firstName = prompt("");
    console.log("Entrez le nom : ");
    const lastName = prompt("");
    const dateOfBirth = UtilsMethod.askDate();
    console.log("Entrez le lieu de naissance : ");
    const placeOfBirth = prompt("");
    console.log("Entrez le numéro d'identité national : ");
    const idNumber = prompt("");

    return new CIN(
      lastName,
      firstName,
      dateOfBirth,
      placeOfBirth,
      idNumber,
      "CIN",
    );
  }

  // Accessor methods (getters and setters)
  getLastName() {
    return this.lastName;
  }

  setLastName(lastName) {
    this.lastName = lastName;
  }

  getFirstName() {
    return this.firstName;
  }

  setFirstName(firstName) {
    this.firstName = firstName;
  }

  getDateOfBirth() {
    return this.dateOfBirth;
  }

  setDateOfBirth(dateOfBirth) {
    this.dateOfBirth = dateOfBirth;
  }

  getPlaceOfBirth() {
    return this.placeOfBirth;
  }

  setPlaceOfBirth(placeOfBirth) {
    this.placeOfBirth = placeOfBirth;
  }

  getIdNumber() {
    return this.idNumber;
  }

  setIdNumber(idNumber) {
    this.idNumber = idNumber;
  }

  toString() {
    return `CIN{lastName='${this.lastName}', firstName='${this.firstName}', dateOfBirth=${this.dateOfBirth}, placeOfBirth='${this.placeOfBirth}', idNumber='${this.idNumber}'}`;
  }
}

module.exports = CIN;

File: CIN.py
from UniqueItem import UniqueItem
from UtilsMethod import UtilsMethod


class CIN(UniqueItem):
    def __init__(self, last_name, first_name, date_of_birth, place_of_birth, id_number, name):
        super().__init__(name)
        self.last_name = last_name
        self.first_name = first_name
        self.date_of_birth = date_of_birth
        self.place_of_birth = place_of_birth
        self.id_number = id_number

    @classmethod
    def from_user_input():
        print("Entrez le prénom : ")
        first_name = input()
        print("Entrez le nom : ")
        last_name = input()
        date_of_birth = UtilsMethod.ask_date(input)
        print("Entrez le lieu de naissance : ")
        place_of_birth = input()
        print("Entrez le numéro d'identité national : ")
        id_number = input()

        return CIN(last_name, first_name, date_of_birth, place_of_birth, id_number, "CIN")

    def __str__(self):
        return (
            f"CIN{{lastName='{self.last_name}', firstName='{self.first_name}', "
            f"dateOfBirth={self.date_of_birth}, placeOfBirth='{self.place_of_birth}', "
            f"idNumber='{self.id_number}'}}"
        )
